src/state/StateNode.java

```diff
@@ -174,4 +159,37 @@ public int getIdentifier() {
     public void setIdentifier(int id) {
         this.id = id;
     }
+
+    private String nodeToString() {
```

Comment:
The naming of this method clashes with the expected behaviour of `toString`. Maybe name this something like `getStateString`, to make it clear that this method returns the string representation of the state without the name of the node.